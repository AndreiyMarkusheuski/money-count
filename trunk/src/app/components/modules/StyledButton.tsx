import React from "react";

type Props = {
    textButton: string,
    handleClick: () => void
}

export const StyledButton = ({ textButton, handleClick }: Props): JSX.Element => {
    return <button
        onClick={() => handleClick()}
    >
        {textButton}
    </button>

}