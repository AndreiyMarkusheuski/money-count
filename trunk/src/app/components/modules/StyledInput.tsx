import React from "react";

type Props = {
    typeInput: string,
    placeholder: string,
    valueInput: string,
    handleChange: (e: any) => void
}

const StyledInput = ({ typeInput, placeholder, valueInput, handleChange }: Props) => {
    return (
        <input
            type={typeInput}
            placeholder={placeholder}
            value={valueInput}
            onChange={handleChange}
            data-testid="add-word-input"
        ></input>
    )
}

export default StyledInput;