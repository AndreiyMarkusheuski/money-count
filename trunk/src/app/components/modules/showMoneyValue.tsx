import React from 'react';

type IShowMoneyValue = {
  moneyValue : number
}

const ShowMoneyValue = (props : IShowMoneyValue) => {
  return (
    <div className="block-money">
      <p>Текущий счет</p>
      <h2>
        <b className="block-money_high">{props.moneyValue}</b> руб
      </h2>
    </div>
  );
};

export default ShowMoneyValue;