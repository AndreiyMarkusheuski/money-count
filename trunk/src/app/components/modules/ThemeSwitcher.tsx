import React, {useContext} from "react";
import { ThemeContext } from "../../context";
import { StyledButton } from "./StyledButton";


export const ThemeSwitcher = () : JSX.Element => {
    const {isLightTheme, handleChangeTheme} = useContext(ThemeContext);
    return <StyledButton textButton={'switch theme'} handleClick={handleChangeTheme}/>
}