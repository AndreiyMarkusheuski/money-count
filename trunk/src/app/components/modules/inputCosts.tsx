import React, { useState, useReducer, useEffect } from "react";
import {StyledButton} from  './StyledButton';
import StyledInput from './StyledInput'
import initState from '../../reducer/initState';
import reducer from '../../reducer/reducer'

type IInputCosts = {
  handleClick : (inputValue : number) => void
}

const InputCosts = (props : IInputCosts) => {
  const [inputValue, setValue] = useState<string>('');
  const [{count}, dispatch] = useReducer(reducer, initState)

  return (
    <>
    <div className="block-input">
      <StyledInput
        typeInput={'number'}
        placeholder={"введи расходы..."}
        valueInput={inputValue}
        handleChange = {(e: any) => setValue(e.target.value)}
        />
      <StyledButton
        handleClick={() => {
          props.handleClick(parseFloat(inputValue));
          dispatch({type: 'increment'})
          setValue('');
        }}
        textButton = {'OK'}
      />
    </div>

    <p>Количество введенных затрат: {count}</p>
    </>
  );
};

export default InputCosts;
