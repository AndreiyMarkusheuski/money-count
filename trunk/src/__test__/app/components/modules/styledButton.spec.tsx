import React from 'react';
import { render, RenderResult, fireEvent } from '@testing-library/react';
import { StyledButton } from '../../../../app/components/modules/StyledButton'

let documentBody: RenderResult;

const testFunc = () => {
    console.log('test')
}

describe('<StyledButton />', () => {
    beforeEach(() => {
      documentBody = render(<StyledButton textButton={'test'} handleClick={testFunc}/>);
    });

    it('shows text message', () => {
      expect(documentBody.getByText('test')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
      const { baseElement } = documentBody;
      expect(baseElement).toMatchSnapshot();
      expect(documentBody.getByText('test')).toBeInTheDocument();
      fireEvent.click(documentBody.getByText('test'))
      expect(baseElement).toMatchSnapshot();
    });
  });