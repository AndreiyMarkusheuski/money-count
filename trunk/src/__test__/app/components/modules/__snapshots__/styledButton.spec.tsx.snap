// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> matches snapshot 1`] = `
<body>
  <div>
    <button>
      test
    </button>
  </div>
</body>
`;

exports[`<StyledButton /> matches snapshot 2`] = `
<body>
  <div>
    <button>
      test
    </button>
  </div>
</body>
`;
